### PR TITLE
Fix iOS Safari Font Issue

### PR DESCRIPTION
**Pull request by 4855062503**
Basically the dart `google_fonts` package is not suitable for iOS Safari, so it's best to import our own font into a fonts folder instead.

Maybe we could look at removing this package later?

Have tested the committed changes running locally on an iOS simulator, which had the fonts problem before the changes committed in this PR. After the changes, the fonts shown were consistent with Chrome's fonts.

### CODE REVIEW

**Comment by djpnewton** [comment]
Have you got a link to the issue?

**Comment by 4855062503** [comment]
[There were some people having similar problems in 2019](https://github.com/flutter/flutter/issues/41483).

But my fix came just by debugging:

- Checked network tab from phone debugging, google fonts loaded in Chrome browser (Desktop), Safari (Desktop), just not iOS Safari.
- Tried to change from GoogleFonts Poppin to a different GoogleFonts font, nothing changed
- Removed GoogleFonts instance, and replaced with local folder `fonts` font. Worked.

**Comment by djpnewton** [comment]
It looks like you can just add the font to the assets and not change the dart code https://pub.dev/packages/google_fonts#font-bundling-in-assets

**Comment by 4855062503** [comment]
Will test this in the iOS simulator when I get home (left mac at home).

**Comment by 4855062503** [comment]
OK, have tested adding:

# To `pubspec.yaml`:
```yaml
assets:
  - google_fonts/Poppins-Regular.ttf
```
To `theme` in `lib/main.dart`

```dart
textTheme: GoogleFonts.poppinsTextTheme(),
```
This font loads correctly in my iOS simulator.

Is the font to be used Oxygen or Poppins?

**Comment by djpnewton** [comment]
we are using poppins (see config.dart)

so it looks like you can just add the font to the assets:

and also add a note why we need to package the font directly rather then just letting the google package download it

```
assets:
  - google_fonts/Poppins-Regular.ttf
```

you can get rid of the other code changes no?

**Comment by 4855062503** [comment]
> you can get rid of the other code changes no?

If I don't manually add 

```dart
textTheme: GoogleFonts.poppinsTextTheme(),
```

then the fonts don't load in iOS Safari

File: lib/main.dart [hunk 1]
```diff
@@ -58,8 +59,8 @@ class MyApp extends StatelessWidget {
       appBarTheme: AppBarTheme(centerTitle: true, color: ZapSecondary),
       // ignore: deprecated_member_use
       accentColor: ZapPrimary,
+      textTheme: GoogleFonts.poppinsTextTheme(),
```
Comment (djpnewton) [inline lib/main.dart:62]:
```suggestion
      textTheme: ZapTextThemer(),
```

File: lib/main.dart [hunk 2]
```diff
@@ -5,6 +5,7 @@ import 'package:logging/logging.dart';
 import 'package:universal_platform/universal_platform.dart';
 import 'package:universal_html/html.dart' as html;
 import 'package:decimal/decimal.dart';
+import 'package:google_fonts/google_fonts.dart';
```
Comment (djpnewton) [inline lib/main.dart:8]:
```suggestion
```

File: pubspec.yaml [hunk 3]
```diff
@@ -69,29 +69,5 @@ flutter:
...
-  #         style: italic
-  #   - family: Trajan Pro
-  #     fonts:
-  #       - asset: fonts/TrajanPro.ttf
-  #       - asset: fonts/TrajanPro_Bold.ttf
-  #         weight: 700
-  #
-  # For details regarding fonts from package dependencies,
-  # see https://flutter.dev/custom-fonts/#from-packages
+    # have to manually import google_fonts
```
Comment (djpnewton) [inline pubspec.yaml:72]: you should explain *why* we need to manually package the font (eg. `if we do not package the font manually the google_fonts module fails to download the fonts on ios/safari see bug at https://xxx...`)

**Review by djpnewton — CHANGES_REQUESTED** [review]

**Review by djpnewton — APPROVED** [review]
does it work?

**Comment by 4855062503** [comment]
> does it work?

Yup.

**Comment by djpnewton** [comment]
reverted changes because of font color side effects
image
image
disapointing this didnt get caught in testing

**Pull request merged**